src/topology/continuous_function/bounded.lean

```diff
@@ -1030,4 +1032,64 @@ module over the algebra of bounded continuous functions from `α` to `𝕜`. -/
 
 end normed_algebra
 
+section cstar_ring
+/-!
+### Star structure
+
+In this section, if `β` is a C⋆-ring, then the space of bounded
+continuous functions from `α` to `β` inherits a C⋆-ring structure also,
+by using the star operation pointwise. Moreover, if `𝕜` is normed field
+and a ⋆-ring, over which `β` is a normed algebra and a star module, then
+the space of bounded continuous functions from `α` to `β` is a star module
+over `𝕜`. In other words, `α →ᵇ β` is a C⋆-algebra (completeness is
+guaranteed when `β` is complete). -/
+
+variables {𝕜 : Type*} [normed_field 𝕜] [star_ring 𝕜]
+variables [topological_space α] [normed_ring β] [star_ring β] [cstar_ring β]
+variables [normed_algebra 𝕜 β] [star_module 𝕜 β]
+
+/-
+Here we can get away with `[normed_star_monoid β]` instead of `[cstar_ring β]`.
+Then change would be to replace :
+  `cstar_ring.to_normed_star_monoid.norm_star`
+with :
+  `(@normed_star_monoid.norm_star _ _ _ _)`
+But since we are mainly focused on showing that `α →ᵇ β` is a `cstar_ring`, we
+just do the slightly less general thing in favor of simplifying `variables`
+declarations.
+-/
+instance : star_ring (α →ᵇ β) :=
+{ star := λ f,
+    have iso : isometry (star : β → β), from
+      star_add_equiv.to_add_monoid_hom.isometry_of_norm cstar_ring.to_normed_star_monoid.norm_star,
```

Comment:
I think it's worth adding this (i.e. the fact that `star` is an isometry) as a separate lemma about `normed_star_monoid` in `normed_space/star`. Also we might as well do the general case here -- I suspect the result will be shorter than the comment that explains that we're not doing it...

Reply:
You don't want to do this in addition to the `star_normed_group_hom` Heather mentioned below, right?

Reply:
I guess it's not really needed anymore (though it wouldn't hurt to put it in anyway). What about only requiring `[normed_star_monoid β]` instead of `[cstar_ring β]` in the `star_ring` instance? I'm pretty sure the proof doesn't even have to change now.

Reply:
Perhaps you can have `star_add_monoid` and `normed_star_monoid` instances for
```lean
variables [topological_space α] [normed_monoid β] [star_add_monoid β] [normed_star_monoid β]
```

Reply:
I changed the variable declarations to (what I think are) the minimal ones and proved that `star` is an isometry (in `analysis/normed_space/star`. Should we rename it (currently `star_isometry`) and also `star_normed_group_hom` to put it inside the `star` namespace (if we have one of those) so we have access to projection notation? I'm still not clear on exactly how/when that notation is available or applies. A pointer to the right literature would be appreciated.

Reply:
What projection notation are you expecting / hoping to have access to? I don't think there's any reason to change `star_` to `star.` in these names.

Reply:
See, that's what I don't entirely understand, hence why my question likely sounds silly. I'm just not sure exactly when it can be applied. My thinking was to have a sort of "reverse coercion," where `star.nomed_group_hom` would return not the bare function `star` but the bundled `normed_group_hom`.

Reply:
Maybe the point is that's no different than what is already there. I guess that makes sense.

Reply:
You can have a look at the reference manual [here](https://leanprover.github.io/reference/other_commands.html#other-commands) for more info on dot notation. The upshot is that if you have a lemma/def `foo.bar (t : foo)`, then for some `x : foo`, you can use `x.bar` to mean `foo.bar x`. This won't work for something like `star`, which can be applied to elements of any type with the right typeclasses.